### ISSUE TITLE
Support host side cgroups v2

### ISSUE THREAD

**Issue by jodh-intel** [kata-containers/kata-containers#927]
# Problem

The runtime doesn't support cgroups v2 on the host.

```sh
$ egrep -r  "\<cgroupsLoadFunc\(|parentCgroup\(" | grep '[12]'
virtcontainers/container.go:    cgroup, err := cgroupsLoadFunc(cgroups.V1,
virtcontainers/container.go:    parent, err := parentCgroup(cgroups.V1, c.state.CgroupPath)
virtcontainers/container.go:    cgroup, err := cgroupsLoadFunc(cgroups.V1,
virtcontainers/sandbox.go:      cgroup, err := cgroupsLoadFunc(V1Constraints, cgroups.StaticPath(s.state.CgroupPath))
```

# Consequence

This means we cannot run on [F31 and F32, which default to cgroups v2 *only*](https://fedoraproject.org/wiki/Releases/31/ChangeSet#Modify_Fedora_31_to_use_CgroupsV2_by_default) without the user performing surgery on their system (reverting to cgroups v1 by changing bootloader stuff and rebooting - yikes!)

Related:

- https://github.com/kata-containers/kata-containers/pull/739#issuecomment-706647393
- https://github.com/kata-containers/kata-containers/pull/739
- https://github.com/kata-containers/kata-containers/issues/538

**Comment by ariel-adam** [comment]
@jodh-intel was this already fixed?

**Comment by kendfinger** [comment]
This could be more important very soon. Ubuntu 21.10 is going to ship with cgroups v2 enabled by default: https://www.phoronix.com/scan.php?page=news_item&px=Ubuntu-21.10-systemd-cgroup

**Comment by itplayer-de** [comment]
Is there any news or a roadmap here? Also Debian 11 "bullseye" now has cgroups v2 as default. Containerd runs in a "mountpoint does not exist" (also with setting sandbox_cgroup_only=true). Thx & regards

**Comment by zvonkok** [comment]
Arch and all derivates (EndeavourOS, Manjaro, Garuda) have also per default cgroup v2.

**Comment by liubin** [comment]
@jodh-intel we are joining a summer code and can I assign this one to others if you are not started yet?

**Comment by jodh-intel** [comment]
Hi @liubin - yes, please do! ;) I was not involved in discussions on this being assigned to me, and don't have the time to look at it.

**Comment by bradennapier** [comment]
:( Manjaro no bueno due to this it seems!

**Comment by jodh-intel** [comment]
This issue should have been closed when #4397 landed (issue #3073).

See: https://github.com/kata-containers/kata-containers/blob/main/docs/design/host-cgroups.md

Closing this issue, but I've raised #6259 to fix `kata-manager` which by default fails (incorrectly) when run on a cgroupsv2 system.
